keras_cv/layers/object_detection_3d/heatmap_decoder.py

```diff
@@ -138,65 +141,67 @@ def __init__(
         self.spatial_size = spatial_size
         self.built = True
 
-    def call(
-        self, prediction: tf.Tensor
-    ) -> Tuple[tf.Tensor, tf.Tensor, tf.Tensor]:
+    def call(self, prediction):
         """Accepts raw predictions, and returns decoded boxes.
 
         Args:
             prediction: float Tensor.
         """
-        heatmap = tf.nn.softmax(prediction[..., :2])[..., 1:2]
-        heatmap_pool = tf.nn.max_pool2d(heatmap, self.max_pool_size, 1, "SAME")
+        heatmap = ops.softmax(prediction[..., :2])[..., 1:2]
+        heatmap_pool = ops.max_pool(heatmap, self.max_pool_size, 1, "same")
         heatmap_mask = heatmap > self.heatmap_threshold
-        heatmap_local_maxima_mask = tf.math.equal(heatmap, heatmap_pool)
+        heatmap_local_maxima_mask = ops.equal(heatmap, heatmap_pool)
         # [B, H, W, 1]
-        heatmap_mask = tf.math.logical_and(
-            heatmap_mask, heatmap_local_maxima_mask
-        )
+        heatmap_mask = ops.logical_and(heatmap_mask, heatmap_local_maxima_mask)
         # [B, H, W, 1]
-        heatmap = tf.where(heatmap_mask, heatmap, 0)
+        heatmap = ops.where(heatmap_mask, heatmap, 0)
         # [B, H, W]
-        heatmap = tf.squeeze(heatmap, axis=-1)
+        heatmap = ops.squeeze(heatmap, axis=-1)
 
         b, h, w = voxel_utils.combined_static_and_dynamic_shape(heatmap)
-        heatmap = tf.reshape(heatmap, [b, h * w])
-        _, top_index = tf.math.top_k(heatmap, k=self.max_num_box)
+        heatmap = ops.reshape(heatmap, [b, h * w])
+        _, top_index = ops.top_k(heatmap, k=self.max_num_box)
 
         # [B, H, W, ?]
         box_prediction = prediction[:, :, :, 2:]
-        f = box_prediction.get_shape().as_list()[-1]
-        box_prediction = tf.reshape(box_prediction, [b, h * w, f])
-        heatmap = tf.reshape(heatmap, [b, h * w])
+        f = list(box_prediction.shape)[-1]
```

Comment:
nit: I don't think that the list cast does anything, you can index into a shape on all backends.